src/client/store/data/customLayers.js

```diff
@@ -0,0 +1,41 @@
+import randomColor from '../../lib/random-color';
+import { getRealApiData } from "../../lib/get-api-data";
+export const state = () => []
+
+export const mutations = {
+  addLayer(state, layer) {
+    state.push(layer)
+  },
+}
+
+export const actions = {
+  async getCustomMapLayers({ commit, dispatch }, parameters) {
+    const body = {
+      url: parameters.serverUrl,
+      type: parameters.serverType,
+    }
+    const data = await getRealApiData('maplayers', body)
```

Comment:
`getRealApiData` is an implementation detail, should be an other method available